src/app/util/attribute-storage.cpp

```diff
@@ -878,6 +878,51 @@ static uint16_t findIndexFromEndpoint(EndpointId endpoint, bool ignoreDisabledEn
     return kEmberInvalidEndpointIndex;
 }
 
+uint16_t emberAfGetClusterServerEndpointIndex(EndpointId endpoint, ClusterId cluster, uint16_t fixedClusterServerEndpointCount)
+{
+    uint16_t epIndex = findIndexFromEndpoint(endpoint, true /*ignoreDisabledEndpoints*/);
+
+    // Endpoint must be configured and enabled
+    if (epIndex != kEmberInvalidEndpointIndex)
+    {
+        if (emberAfFindClusterInType(emAfEndpoints[epIndex].endpointType, cluster, CLUSTER_MASK_SERVER) == nullptr)
+        {
+            // The provided endpoint do not contain the given cluster server.
+            return kEmberInvalidEndpointIndex;
+        }
+
+        if (epIndex < FIXED_ENDPOINT_COUNT)
+        {
+            // This endpoint is a fixed one.
```

Comment:
Isn't this just reimplementing `emberAfFindClusterServerEndpointIndex`?  Why not just call that?

Reply:
While true, this approach duplicates various fragile code.

Perhaps as a followup (or here, if you prefer) we should factor out the for loop inside `findClusterEndpointIndex` so that it can be used from both there and here?

Reply:
I can do that. I just wonder once this is merged and tested with a unit test. Is there still a use case for emberAfFindClusterServerEndpointIndex ?

Reply:
That's a good question.  There are a bunch of consumers, but I suspect they should all be using the new function...  So maybe the right answer is we update them all to do that and remove `emberAfFindClusterServerEndpointIndex`?

Reply:
Yes, I think that's a plan. In a second PR, I'll review all the other use of `emberAfFindClusterServerEndpointIndex` and replace it with `emberAfGetClusterServerEndpointIndex`